server/printing/printing_server.py

```diff
@@ -36,7 +37,9 @@ def SendRequestToPrinter(self, encoded_file, copies=1, options={}):
         command += "tmp.pdf"
         status = os.popen(command)
         os.remove("tmp.pdf")
-        return 'printed' if status else 'error'
+        mock = SCEPrinter()
+        mock.SCEPrinter(self.DeterminePrinterForJob)
```

Comment:
This is a good start! but you gotta make use of the methods you created in your other file.

for example, when sending a print request, your mock needs to update! you determine which printer to use, you need to pass that printer to your mock so it knows which printer to mock as well.

consider adding this to `SendRequestToPrinter`
```
   
        printer = self.DeterminePrinterForJob(copies)
        command += "-d " + printer + " "
        command += "tmp.pdf"
        mock.get_printer(printer)
        status = os.popen(command)
        os.remove("tmp.pdf")
        return 'printed' if status else 'error'
```

then once you know which printer you're working with you can get the copies and pages into your mock as well